app/scripts/metamask-controller.js

```diff
@@ -2702,6 +2809,19 @@ export default class MetamaskController extends EventEmitter {
         ),
       ///: END:ONLY_INCLUDE_IN
 
+      // AccountsController
+      setSelectedInternalAccount:
+        accountsController.setSelectedAccount.bind(accountsController),
+
+      setAccountName:
+        accountsController.setAccountName.bind(accountsController),
+
+      setAccountLabel: (address, label) => {
+        this.preferencesController.setAccountLabel(address, label);
+        const account = this.accountsController.getAccountByAddress(address);
+        this.accountsController.setAccountName(account.id, label);
```

Comment:
It looks like `account` could be `undefined` here, do we need to check it?

``` suggestion
        if (account) {
          this.accountsController.setAccountName(account.id, label);
        }
```

Or do we need to throw an error?

``` suggestion
        if (account) {
          this.accountsController.setAccountName(account.id, label);
        } else {
          throw new Error("Could not find account");
        }
```

Reply:
Throwing an error sounds appropriate for this case. Done in [957a205](https://github.com/MetaMask/metamask-extension/pull/21437/commits/957a205fe9e0a800c0b8bbff8ac3e314456ace88)